### PR TITLE
Pass empty as add item template GUID instead of project GUID in AbstractAddClassProjectCommand

### PR DESCRIPTION
**Pull request by davkean**
- Pass empty as add item template GUID instead of project

AddItemDialog isn't expecting the project GUID - it's expecting the Add Item Template GUID. We pass empty here - as it ends up asking the hierarchy for it, see CAddProjectItemDlg::AddProjectItemDlgTitledEx.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system @333fred

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Input/Commands/AbstractAddClassProjectCommandTests.cs [hunk 1]
```diff
@@ -224,10 +215,10 @@ public void GetCommandStatusAsync_NullAsNodes_ThrowsArgumentNull()
 
         internal abstract string DirName { get; }
 
-        internal AbstractAddClassProjectCommand CreateInstance(IPhysicalProjectTree projectTree = null, IUnconfiguredProjectVsServices projectVsServices = null, Shell.SVsServiceProvider serviceProvider = null, IProjectTreeProvider provider = null, IVsAddProjectItemDlg dlg = null, Func<ProjectProperties> properties = null)
+        internal AbstractAddClassProjectCommand CreateInstance(IPhysicalProjectTree projectTree = null, IUnconfiguredProjectVsServices projectVsServices = null, Shell.SVsServiceProvider serviceProvider = null, IProjectTreeProvider provider = null, IVsAddProjectItemDlg dlg = null)
         {
             projectTree = projectTree ?? IPhysicalProjectTreeFactory.Create(provider);
-            projectVsServices = projectVsServices ?? IUnconfiguredProjectVsServicesFactory.Implement(projectProperties: properties);
+            projectVsServices = projectVsServices ?? IUnconfiguredProjectVsServicesFactory.Implement(threadingServiceCreator: () => IProjectThreadingServiceFactory.Create());
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Input/Commands/AbstractAddClassProjectCommandTests.cs]: you don't need to pass `threadingServiceCreator`

**Comment by basoundr** [comment]
[Here](https://devdiv.visualstudio.com/DevDiv/_git/VS?path=%2Fsrc%2Fenv%2Fmsenv%2Fcore%2Fvsprojui.cpp&version=GBlab%2Fd15prerel&_a=contents&line=1460&lineStyle=plain&lineEnd=1461&lineStartColumn=1&lineEndColumn=1) we try to get the AddItemTemplateGuid and if we fail to get it then we use the Project type Guid. We should not be calling into API based on the implementation. It is also not a costly operation to get properties given that it is a user triggered operation.

**Comment by davkean** [comment]
@basoundr The project GUID is not the correct GUID here. They are expecting a _project type_ GUID, or in our case, our our _Add Item Template_ GUID. Retrieving the _Add Item Template_ GUID is exactly what the dialog is doing, and I do not consider it an implementation detail given it's observable GetProperty on the hierarchy.

**Comment by basoundr** [comment]
@davkean Alright then

**Pull request merged**